r/src/compute.cpp

```diff
@@ -398,6 +398,21 @@ std::shared_ptr<arrow::compute::FunctionOptions> make_compute_options(
     return out;
   }
 
+  if (func_name == "mode") {
+    using Options = arrow::compute::ModeOptions;
+    auto out = std::make_shared<Options>(Options::Defaults());
+    if (!Rf_isNull(options["n"])) {
+      out->n = cpp11::as_cpp<int>(options["n"]);
```

Comment:
`ModeOptions::n` is an `int64_t`, so perhaps use `cpp11::as_cpp<int64_t>`?

Reply:
Cheers for spotting that, will update them both!